velox/vector/VariantToVector.cpp

```diff
@@ -77,4 +77,165 @@ ArrayVectorPtr variantArrayToVector(
       pool);
 }
 
+// Structure used to keep check on data type of
+// the complex vector, along with the number and data
+// types of constituent elements. Attributes elementCount
+// and elementInserted keeps the length of the vector, and
+// the number of elements inserted during the operation
+// respectively.
+struct NestedTypeCounter {
+  TypeKind kind = TypeKind::UNKNOWN;
+  vector_size_t elementCount = 0;
+  vector_size_t elementsInserted = 0;
+  std::vector<NestedTypeCounter> children;
```

Comment:
We should move this structure and this code to inside pyvelox as vector code is senstive and has large dependencies inside meta. Secondly it seems unsafe to have a struct being mutated over multiple passes. Its better to restructure it so that its constructed over one pass. 
Finally if you do want to take multiple passes, its better to actually construct the type , which itself is a tree structure (encompassing typekinds) over this.

Reply:
Moved it to `complex.cpp` and `complex.h` in pyvelox directory, and now using `Type` to construct the tree.